### PR TITLE
test: overhaul tests to support `make test` & minimal init

### PR DESCRIPTION
**Pull request by PriceHiller**
Copied from the body of the commit message:

> This now allows tests to be ran via `make test`.
> 
> Currently the tests defined in `main_spec.lua` are invalid, as the `run_file` method appears to have been removed at some point.
> 
> At the bare minimum though, this commit gives `rest.nvim` a decent framework to work off of for running tests in the future when written.

As noted in the body of the commit -- the tests currently written don't work due to a removal of a function in the past.

Hopefully this eases writing tests down the road if someone decides to write them.

The minimal init tests do expect whoever is running them to have `luarocks` installed -- it directly calls into the `luarocks` cli as an fyi.

If you don't care about the tests/don't want the maintenance burden on this, feel free to close this out 🙂.

### CODE REVIEW

File: tests/minimal_init.lua [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+---Make the `rest.nvim` lua rock and register the resulting plugin on the `runtimepath`
+function M.make_luarock()
+  print(">> Making rest.nvim luarock")
+  local luarocks_binary = "luarocks"
+  local install_destination = M.root("rocks/")
+  vim.g.rocks_nvim = {
+    rocks_path = install_destination,
+    luarocks_binary = luarocks_binary,
+  }
```
Comment (NTBBloodbath) [inline tests/minimal_init.lua:60]: Is this here as leftovers from a past changes iteration (or something around these lines) or do you plan to use `rocks.nvim` later for testing? 👀

Comment (PriceHiller) [reply tests/minimal_init.lua:60]: Nope, that's some leftovers 😅.

I went down the *horrid*, ***horrid***, road of not rolling my own and using other plugins to handle the luarocks stuff. They're all *way* too immature lacking basic stuff to just synchronously install a plugin.

I ended up rolling a super simple wrapper instead to call out to luarocks.

I'll remove it.

Comment (PriceHiller) [reply tests/minimal_init.lua:60]: I may end up PR'ing to `rocks.nvim` to fix the issues I ran into with it when it came to creating a test harness.

Comment (PriceHiller) [reply tests/minimal_init.lua]: Removed as part of the force push from `bbcf8ea` -> `45f9046`

**Comment by NTBBloodbath** [comment]
> As noted in the body of the commit -- the tests currently written don't work due to a removal of a function in the past.
> 
> Hopefully this eases writing tests down the road if someone decides to write them.

Oh yeah, I was planning on adjusting them again later and this will really make the process easier. Before there was a workflow with Nix, but I know absolutely nothing about Nix and there was no one to maintain the flake so I decided to remove it, and when the tests are functional again I will make a testing CI using the Makefile :)

> The minimal init tests do expect whoever is running them to have `luarocks` installed -- it directly calls into the `luarocks` cli as an fyi.

That's fine with me, could we mention it somewhere in the readme (maybe under the Contributing section)?

> If you don't care about the tests/don't want the maintenance burden on this, feel free to close this out 🙂.

It's fine, tests are very important to me. I wanted to give it its update due to the tests but I did not want to continue delaying the release of v2 (it was already almost 2 months longer than I planned 😅)

**Comment by PriceHiller** [comment]
> Oh yeah, I was planning on adjusting them again later and this will really make the process easier. Before there was a workflow with Nix, but I know absolutely nothing about Nix and there was no one to maintain the flake so I decided to remove it, and when the tests are functional again I will make a testing CI using the Makefile :)

Nix is damn cool, I use it a lot here locally. Unfortunately, there's not a learning curve to it, there's a learning abyss -- not exactly user friendly lol. Don't blame you at all for yanking it.

> That's fine with me, could we mention it somewhere in the readme (maybe under the Contributing section)?

Updated the README to mention this, let me know if you want me to change the wording.

**Review by NTBBloodbath — APPROVED** [review]

**Comment by NTBBloodbath** [comment]
> > Oh yeah, I was planning on adjusting them again later and this will really make the process easier. Before there was a workflow with Nix, but I know absolutely nothing about Nix and there was no one to maintain the flake so I decided to remove it, and when the tests are functional again I will make a testing CI using the Makefile :)
> 
> Nix is damn cool, I use it a lot here locally. Unfortunately, there's not a learning curve to it, there's a learning abyss -- not exactly user friendly lol. Don't blame you at all for yanking it.

Yeah, hopefully one day the use of Nix will be able to resume as it is really powerful :p

> > That's fine with me, could we mention it somewhere in the readme (maybe under the Contributing section)?
> 
> Updated the README to mention this, let me know if you want me to change the wording.

It looks good to me, thank you! 🙂

**Pull request merged**